hr_holidays_public/models/hr_holidays_public.py

```diff
@@ -86,50 +90,69 @@ def get_holidays_list(
         :param employee_id: ID of the employee
         :return: recordset of hr.holidays.public.line
         """
+        partner = self._get_partner_deprecated_employee(partner_id, employee_id)
+
         if not start_dt and not end_dt:
             start_dt = datetime.date(year, 1, 1)
             end_dt = datetime.date(year, 12, 31)
         years = list(range(start_dt.year, end_dt.year + 1))
         holidays_filter = [("year", "in", years)]
-        employee = False
-        if employee_id:
-            employee = self.env["hr.employee"].browse(employee_id)
-            if employee.address_id and employee.address_id.country_id:
-                holidays_filter.append("|")
-                holidays_filter.append(("country_id", "=", False))
-                holidays_filter.append(
-                    ("country_id", "=", employee.address_id.country_id.id)
-                )
-            else:
-                holidays_filter.append(("country_id", "=", False))
+        if partner.country_id:
+            holidays_filter.append("|")
+            holidays_filter.append(("country_id", "=", False))
+            holidays_filter.append(("country_id", "=", partner.country_id.id))
+        else:
+            holidays_filter.append(("country_id", "=", False))
         pholidays = self.search(holidays_filter)
         if not pholidays:
             return self.env["hr.holidays.public.line"]
 
         states_filter = self._get_domain_states_filter(
-            pholidays, start_dt, end_dt, employee_id
+            pholidays, start_dt, end_dt, partner_id=partner.id
         )
         hhplo = self.env["hr.holidays.public.line"]
         holidays_lines = hhplo.search(states_filter)
         return holidays_lines
 
     @api.model
-    def is_public_holiday(self, selected_date, employee_id=None):
+    def is_public_holiday(self, selected_date, employee_id=None, partner_id=None):
         """
         Returns True if selected_date is a public holiday for the employee
         :param selected_date: datetime object
         :param employee_id: ID of the employee
+        :param partner_id: ID of the partner
         :return: bool
         """
+        partner = self._get_partner_deprecated_employee(partner_id, employee_id)
         holidays_lines = self.get_holidays_list(
-            year=selected_date.year, employee_id=employee_id
+            year=selected_date.year, partner_id=partner.id
         )
         if holidays_lines:
             hol_date = holidays_lines.filtered(lambda r: r.date == selected_date)
             if hol_date.ids:
                 return True
         return False
 
+    def _get_partner_deprecated_employee(self, partner_id, employee_id):
+        # TODO: Drop function in next migration
+        employee = False
+        partner = False
+        if employee_id is not None:
```

Comment:
what if you call this method with `(False, False)` ?

Reply:
From where it's called, it's supposed to have a value or be None, so no need to handle such a case